lib/utils/optionValidation.ts

```diff
@@ -131,7 +131,7 @@ export function createAudioDeviceValidator(
  * otherwise `undefined`.
  */
 export function validateDeviceId(option: any): Validity {
-  if (option && typeof option !== 'string') {
+  if (!['undefined', 'string'].includes(typeof option)) {
```

Comment:
Would we want to fail or pass on null values? I was thinking that because we define in the `Options` interface the option `deviceId` to be only a `string?`, we wouldn't want to accept anything else.

Also, is an empty string a valid `deviceId` as well? Passing it to `gUM` in all three major browsers seems to work.

Reply:
Let's think of it also outside of TS because our SDK can be used on non-TS environment.
Your new changes will result with the following:
- `deviceId = undefined` -> pass
- `deviceId = null` -> fail
- `deviceId = ""` -> pass

What we want is to treat null, undefined, and empty string the same. You could put `null` in your checks and that should work, but I think `option && typeof option !== 'string'` is simpler.

Reply:
Ah, I recall why I changed this in the first place. This doesn't catch when `option` is `0`.